client/server/lib/performance-mark/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Updates the express request context with a new performance mark.
+ *
+ * This pushes a new "performance mark" object to the array of performance marks.
+ * This object includes the current time to note when this step of the pipeline
+ * started. We also update the previous mark with its duration. (E.g. we expect
+ * each mark to happen in serial, with the previous step ending as the new one begins.)
+ *
+ * Unfortunately, due to how express passes around the request object, this modifies
+ * the request context by reference.
+ *
+ * @param {object}  req      The express request object.
+ * @param {string}  markName A name for the marker being logged.
+ * @param {boolean} isChild  Optionally note this occured as part of a different mark.
+ */
+export default function performanceMark( req, markName, isChild = false ) {
+	if ( ! req.context ) {
+		return;
+	}
+
+	req.context.performanceMarks ??= [];
```

Comment:
Nice to see logical nullish assignment used in the codebase 👍 It's fine to use it here, it's been fully supported since Node 15.

Reply:
It's super helpful!